inc/commontreedropdown.class.php

```diff
@@ -43,6 +43,18 @@ abstract class CommonTreeDropdown extends CommonDropdown {
 
    public $can_be_translated = false;
 
+   /**
+    * @var \Psr\SimpleCache\CacheInterface
+    */
+   protected $cache;
```

Comment:
Why not n `CommonDBTM`?

Reply:
If we inject it in CommonDBTM, it will be hard to know which class really requires this when we will be able to inject service using DI.

Reply:
Well... caching must be available almost everywhere (for read at least); it is hard to predict in which classes we really need it.